cfme/cloud/keypairs.py

```diff
@@ -158,46 +193,22 @@ def refresh():
             self.provider.refresh_provider_relationships()
             view.browser.selenium.refresh()
             view.flush_widget_cache()
-        view = navigate_to(self, 'All')
+        view = navigate_to(self, 'Details')
```

Comment:
**Optional**

Doesn't really have to do with Collection, but consider removing this convenience method and making an enhancement to KeyPair.exists to catch navigation exception and return false.

Then callers can directly call `wait_for(lambda: keypair.exists)`